content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js

```diff
@@ -120,62 +120,72 @@
-                    /*
-                     * Editor message handling:
-                     * - subscribe to "cmp.panelcontainer" message requests sent by the editor frame
-                     * - check that the message data panel container type is correct and that the id (path) matches this specific Accordion component
-                     * - if so, route the "navigate" operation to enact a navigation of the Accordion based on index data
-                     */
-                    window.CQ.CoreComponents.MESSAGE_CHANNEL = window.CQ.CoreComponents.MESSAGE_CHANNEL || new window.Granite.author.MessageChannel("cqauthor", window);
-                    window.CQ.CoreComponents.MESSAGE_CHANNEL.subscribeRequestMessage("cmp.panelcontainer", function(message) {
-                        if (message.data && message.data.type === "cmp-accordion" && message.data.id === that._elements.self.dataset["cmpPanelcontainerId"]) {
-                            if (message.data.operation === "navigate") {
-                                // switch to single expansion mode when navigating in edit mode.
-                                var singleExpansion = that._properties.singleExpansion;
-                                that._properties.singleExpansion = true;
-                                toggle(message.data.index);
-
-                                // revert to the configured state.
-                                that._properties.singleExpansion = singleExpansion;
+        /**
+         * Displays the panel containing the element that corresponds to the deep link in the URI fragment
+         * and scrolls the browser to this element.
+         */
+        function scrollToDeepLinkIdInAccordion() {
+            if (containerUtils) {
+                var deepLinkItemIdx = containerUtils.getDeepLinkItemIdx(that, "item", "item");
+                if (deepLinkItemIdx && deepLinkItemIdx !== -1) {
+                    var deepLinkItem = that._elements["item"][deepLinkItemIdx];
+                    if (deepLinkItem && !deepLinkItem.hasAttribute(dataAttributes.item.expanded)) {
+                        // if single expansion: close all accordion items
+                        if (that._properties.singleExpansion) {
+                            for (var j = 0; j < that._elements["item"].length; j++) {
+                                if (that._elements["item"][j].hasAttribute(dataAttributes.item.expanded)) {
+                                    setItemExpanded(that._elements["item"][j], false, true);
+                                }
                             }
                         }
-                    });
+                        // expand the accordion item containing the deep link
+                        setItemExpanded(deepLinkItem, true, true);
+                    }
+                    var hashId = window.location.hash.substring(1);
+                    var hashItem = document.querySelector("[id='" + hashId + "']");
```

Comment:
should this query not be conditional in case the hashId isn't empty?

Reply:
Addressed in https://github.com/adobe/aem-core-wcm-components/pull/2138/commits/5b6735af3ae0bf75e885c69d9bf51dc94cd643be